### DISCUSSION TITLE
Add translations for measurement units

### DISCUSSION THREAD

**Discussion by autinerd** [home-assistant/architecture#1056] [Entity Models] [answered]
# Context

Currently, we have have ability to translate entity names, but not measurement units, which means that these are all in English. For the most SI units, that's not a problem, but there are occasions where some languages have different names or some integrations have custom units which should be translated.

We already have `translation_key` for entities, this would extend the `strings.json` to have a `native_unit_of_measurement` besides the name or stand-alone, if only the unit of measurement needs to be translated.

Example:
```json
{
  "entity": {
    "sensor": {
      "stargazers_count": {
        "name": "Stars",
        "native_unit_of_measurement": "Stars"
      },
    }
  }
}
```

# Integrations

## General translations

Some measurement units are spelled differently in some languages, therefore all integrations using these would profit.

For example:
- Byte (B), Kilobyte (kB), Megabyte (MB) etc.: In French, these are called octets (o), kilooctets (ko), megaoctets (Mo) etc.

## Specific integrations with custom units of measurement

<details>

<summary>There are also integrations which have custom units of measurements, which are currently in English:</summary>

- `adguard`
- `ambient_station`
- `analytics_insights`
- `ecoforest`
- `ecowitt`
- `fitbit`
- `geocaching`
- `github`
- `guardian`
- `jellyfin`
- `lamarzocco`
- `ld2410_ble`
- `lidarr`
- `netgear_lte`
- `nextdns`
- `nzbget`
- `onewire`
- `permobil`
- `pi_hole`
- `radarr`
- `rainmachine`
- `rfxtrx`
- `roomba`
- `shelly`
- `solaredge_local`
- `sonarr`
- `starline`
- `switchbot`
- `tautulli`
- `withings`
- `youtube`

</details>

These could all profit from measurement unit translations.

**Comment by emontnemery** [comment]
@autinerd `youtube` has a sensor which counts subscribers and `withings` has a sensor which counts steps. Is it the case for all the listed integrations with custom units that they have sensors which represent the count of something?

**Reply by autinerd** [reply]
Yes, they are all representing the amount of something. There are also some integrations which have units as custom which are just not defined in one of the provided enums like "kcal" but these are not meant here.

**Comment by emontnemery** [comment]
@autinerd In most cases we only translate in frontend, the current exceptions are entity and device names which are translated in core. Please clarify your proposal where translation of the unit would happen, would translated units of measurement be frontend only, or would we need to translate the unit in core?

**Reply by autinerd** [reply]
For the general ones, the frontend should be good to go (MB -> Mo etc.)

For the custom ones, I think that it would need to be in core (but to be able to judge it I don't know enough about the internal relation between frontend and core, this thinking is based on the fact that the custom units are defined at the same place as the entity names)

**Comment by emontnemery** [comment, marked as answer]
@autinerd let's go ahead with an implementation of this proposal 👍

Two comments on the implementation:
- We don't want to allow translating standardized unit symbols, that should be clear from the documentation
- Translation should happen in frontend when displaying the state

**Reply by frenck** [reply]
Implemented in https://github.com/home-assistant/core/pull/130538
